### ISSUE TITLE
Lightning app does not start in Ubuntu 16.04. Module not found

### ISSUE THREAD

**Issue by Raulo** [lightninglabs/lightning-app#97]
### Description
The application does not start. Error message below.  

### Steps to reproduce the behavior
1. Install npm on 16.04/x86_64 Ubuntu
$ sudo apt install npm nodejs-legacy
2. Follow the installation instructions from https://github.com/lightninglabs/lightning-app 
3. $ npm start

### Screenshot and/or logs if applicable
`
[webpack] ERROR in $HOME/tmp/lightningnetwork/lightning-app/packages/lightning-store/\~/redux-electron-ipc/index.js
[webpack] Module not found: Error: Cannot resolve 'file' or 'directory' ./dist/electron-redux-ipc.js in $HOME/tmp/lightningnetwork/lightning-app/packages/lightning-store/node_modules/redux-electron-ipc
[webpack]  @ $HOME/tmp/lightningnetwork/lightning-app/packages/lightning-store/~/redux-electron-ipc/index.js 1:17-56
[electron] npm run start-electron exited with code 0
--> Sending SIGTERM to other processes..
[webpack] npm run start-webpack exited with code null

npm ERR! Linux 4.10.0-38-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "start"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! lightning-desktop@1.1.0 start: \`node scripts/startup-message.js && concurrently --kill-others "npm run start-electron" "npm run start-webpack" -n "electron,webpack" -p name\`
`

I guess it is a npm module missing or incompatible version but the required versions are not explained in the documentation.

**Comment by rootMerkle** [comment]
I don’t think this is an issue with the application. Ubuntu’s apt doesn’t have the latest node source. You should install node and npm from nodesource...

curl -sL https://deb.nodesource.com/setup_9.x | sudo -E bash -
sudo apt-get install -y nodejs

Full instructions here:
https://github.com/nodesource/distributions/blob/master/README.md

**Comment by Raulo** [comment]
Still the same problem. 

npm@5.5.1
node@v9.2.0

I have zero experience with npm but I'm willing to help in tracking it down.

**Comment by iraszl** [comment]
I have some basic questions to wrap my head around LN:
1. If I have 3 friends. I have a channel to one of them. And, he has
channels to the remaining 2. Can we all send to each other coins? Or do we
need channels between all of us individually?
2. Within the same group of friends: If we want to be able to send and
receive up to $100. How much BTC needs to be locked up? Is it just $100 per
person? Or more?
3. If I lock up $100, can I immediately receive up to $100, or do I have to
first spend those $100, before I can start receiving?

**Comment by 34ro** [comment]
@Raulo I had same problem.
I removed ^ from redux-electron-ipc version before executing `npm run setup`  for workaround.
The version of npm and node is the same as you.
I think redux-electron-ipc 1.1.11 has a problem.

```
--- a/packages/lightning-store/package.json
+++ b/packages/lightning-store/package.json
@@ -19,7 +19,7 @@
     "react-router-dom": "^4.0.0",
     "react-router-redux": "^5.0.0-alpha.6",
     "redux": "^3.6.0",
-    "redux-electron-ipc": "^1.1.3",
+    "redux-electron-ipc": "1.1.3",
     "redux-grpc-middleware": "^0.1.1",
     "redux-logger": "^2.7.4",
     "redux-selector": "^0.1.0",
```

**Comment by Raulo** [comment]
@34ro,

Thank you it helped. Though, my package.json contained  "redux-electron-ipc": "^1.1.11" string that I replaced for "redux-electron-ipc": "1.1.3".

The GUI started but any operation complained about 
expected 1 macaroon, got 0. After some heavy googling and putting no-macaroons=true into lnd.conf (I have no idea what security implications it has), it seems to work.

Now, if only I could find somebody to send coins to. 

Please, update documentation about installing nodejs and the no-macaroons requirement.

**Comment by bryanvu** [comment]
Hi all, thanks for the issue report. I just tried to replicate this with node v9.2.0 (running with nvm), and I didn't see any of the errors above. I'm thinking that perhaps redux-electron-ipc fixed things on their end. If you see this again, please reopen. Also, after a bit more testing, I'll be pushing a change that will fix the macaroon issue.

**Comment by PaulCapestany** [comment]
FWIW, I also seemingly just ran into the issue, noticed `Uncaught Error: Cannot find module "./dist/electron-redux-ipc.js"` when I went to http://localhost:9997 

@34ro's workaround fixed things for me as well though

**Comment by t4sk** [comment]
**Node**: 9.2.0
**NPM**: 5.5.1

Thanks @34ro!
Your suggestion did work, with some extra steps.
Here are the steps that I took to install and run the app.

```
npm config set save false
npm run clean
npm run setup
```

After installation, `npm start` was throwing an error
```
Main Process: Error: ENOENT: no such file or directory, open '/home/t4sk/.lnd/admin.macaroon'
```

So, in order to create an `admin.macaroon`, I manually ran `lnd` with the following config
```
lnd \
--bitcoin.active \
--bitcoin.testnet \
--neutrino.active \
--neutrino.connect=btcd0.lightning.computer:18333 \
--debuglevel=info \
--autopilot.active \
--noencryptwallet
```

**Comment by wereHamster** [comment]
The problem seems to be that npm v5 (or whatever is bundled with node v8/v9) automatically rewrites your package.json and bumps the lower bounds (WTF is it thinking?). I switched to an older version of npm (the one included in node v6 worked for me). But really, the repository should include a package-lock.json file, to make dependencies deterministic.

redux-electron-ipc v1.1.11 is broken, I already reported it https://github.com/mariotacke/redux-electron-ipc/issues/11
